### PR TITLE
 BaSP-MR-7: superadmin section.

### PR DESCRIPTION
**Pull request by matiasluispeter**
Fix all the routes. Complete the schema for email and password. I did the validations and controllers.

### CODE REVIEW

File: src/controllers/super-admins.js [hunk 1]
```diff
@@ -8,11 +8,104 @@ const getAllsuperAdmins = (req, res) => {
...
+  const { id } = req.params;
+
+  SuperAdmin.findById(id)
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
+          message: `Super admin with id: ${id} was not found`,
+          error: true,
+        });
+      }
```
Comment (LucianoStradiot) [inline src/controllers/super-admins.js]: There should be a validation of the id before it enter to the DB

File: src/controllers/super-admins.js [hunk 2]
```diff
@@ -8,11 +8,104 @@ const getAllsuperAdmins = (req, res) => {
...
+      password,
+    },
+    { new: true },
+  )
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
+          message: `Super admin with id: ${id} was not found`,
+          error: true,
+        });
```
Comment (LucianoStradiot) [inline src/controllers/super-admins.js:74]: Same here and for the delete method

File: src/validations/super-admins.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const Joi = require('joi');
+
+const emailRegex = /^[^\s@]+@[^\s@]+.[^\s@]+$/;
+const passRegex = /^(?=.*\d)(?=.*[A-Z]).{8,}$/;
+
+const validateCreation = (req, res, next) => {
+  const superAdminValidation = Joi.object({
+    email: Joi.string().email().regex(emailRegex).required(),
```
Comment (LucianoStradiot) [inline src/validations/super-admins.js:8]: The email regex validation is not necessary. Only use .email()

File: src/validations/super-admins.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const Joi = require('joi');
+
+const emailRegex = /^[^\s@]+@[^\s@]+.[^\s@]+$/;
+const passRegex = /^(?=.*\d)(?=.*[A-Z]).{8,}$/;
+
+const validateCreation = (req, res, next) => {
```
Comment (LucianoStradiot) [inline src/validations/super-admins.js:6]: The name of this const could be validateSuperCreation for consistency with the other resources of the app

File: src/controllers/super-admins.js [hunk 5]
```diff
@@ -8,11 +8,104 @@ const getAllsuperAdmins = (req, res) => {
...
+};
+const getsuperAdminById = (req, res) => {
+  const { id } = req.params;
+
+  SuperAdmin.findById(id)
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
+          message: `Super admin with id: ${id} was not found`,
+          error: true,
```
Comment (Eugeniagassman) [inline src/controllers/super-admins.js:23]: you should add data: undefinded in case the id is not found.

**Review by LucianoStradiot — APPROVED** [review]
LG!

**Review by fernandofernandezuy — APPROVED** [review]
Looks good to me!

File: src/controllers/super-admins.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    }))
+    .catch((error) => res.status(500).json({
+      message: 'An error was found',
+      error,
+    }));
+};
+const getsuperAdminById = (req, res) => {
+  const { id } = req.params;
+  if (id.length !== 24) {
+    res.status(400).json({
```
Comment (Juanoauil) [inline src/controllers/super-admins.js:18]: you should return this response

File: src/controllers/super-admins.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      message: 'An error was detected',
+      error,
+    }));
+};
+
+const updateSuperAdmin = (req, res) => {
+  const { id } = req.params;
+  const { email, password } = req.body;
+  if (id.length !== 24) {
+    res.status(400).json({
```
Comment (Juanoauil) [inline src/controllers/super-admins.js:68]: Same here

File: src/controllers/super-admins.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        error: false,
+      });
+    })
+    .catch((error) => res.status(500).json(error));
+};
+
+const deleteSuperAdmin = (req, res) => {
+  const { id } = req.params;
+  if (id.length !== 24) {
+    res.status(400).json({
```
Comment (Juanoauil) [inline src/controllers/super-admins.js:101]: Same here

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**